component/main.jsonnet

```diff
@@ -1,10 +1,98 @@
-// main template for openshift4-nodes
+local com = import 'lib/commodore.libjsonnet';
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
-// The hiera parameters for the component
+
 local params = inv.parameters.openshift4_nodes;
 
+local machineSet = function(name, set)
+  local role = if std.objectHas(set, 'role') then set.role else 'worker';
+  kube._Object('machine.openshift.io/v1beta1', 'MachineSet', name)
+  + { spec+: params.defaultSpecs[inv.parameters.cloud.provider] }
+  + {
+    metadata+: {
+      labels+: {
+        'machine.openshift.io/cluster-api-cluster': params.infrastructureID,
+      },
+      namespace: params.namespace,
+    },
+    spec+: {
+      replicas: com.getValueOrDefault(set, 'replicas', 1),
+      selector+: {
+        matchLabels+: {
+          'machine.openshift.io/cluster-api-cluster': params.infrastructureID,
+          'machine.openshift.io/cluster-api-machineset': name,
+        },
+      },
+      template+: {
+        metadata+: {
+          labels+: {
+            'machine.openshift.io/cluster-api-cluster': params.infrastructureID,
+            'machine.openshift.io/cluster-api-machine-role': role,
+            'machine.openshift.io/cluster-api-machine-type': role,
+            'machine.openshift.io/cluster-api-machineset': name,
+          },
+        },
+        spec+: {
+          providerSpec+: {
```

Comment:
```suggestion
          metadata+: {
            labels+: {
              'machine.openshift.io/cluster-api-machine-role': role,
            }
          }
          providerSpec+: {
```

Reply:
This might be the missing part, according to https://docs.openshift.com/container-platform/4.4/machine_management/creating-infrastructure-machinesets.html#machineset-yaml-gcp_creating-infrastructure-machinesets

Reply:
Nope, this does not resolve the issue. Let us move this discussion over to #4.

Reply:
Nevertheless the metadata.labels should be set there, no?

Reply:
Should actually be `['node-role.kubernetes.io/'+role]: ''`